src/centengine/engine_impl.cc

```diff
@@ -189,16 +185,84 @@ grpc::Status engine_impl::GetContact(
 
   command_manager::instance().enqueue(lambda);
   auto promiseresponse = f1.get();
-  
+
   if (promiseresponse.first == true) {
```

Comment:
no need to check if (a == true) because a is a boolean in that case...
So if (a) is better